packages/opentelemetry-sdk-trace-base/src/Tracer.ts

```diff
@@ -40,6 +42,7 @@ export class Tracer implements api.Tracer {
   private readonly _idGenerator: IdGenerator;
   readonly resource: Resource;
   readonly instrumentationLibrary: InstrumentationLibrary;
+  private clocks: WeakMap<api.Span, AnchoredClock> = new WeakMap();
```

Comment:
Why is this needed? Currently nothing is stored in this map but even if you store a span it's mostly useless as there is no guarantee that a parent span was created by the same tracer.

I would either make `span.clock` available or create a map on global scope. As far as I know WeakMaps are not the best thing once it comes to performance therefore a property on span is likely better regarding this.

Reply:
Updated to remove WeakMap